### ISSUE TITLE
PWA Experience?

### ISSUE THREAD

**Issue by MovingGifts** [quantumjs/quantumjs.github.io#1]
Hi @QuantumInformation,

_Apologies for posting here but could not find another way to reach out to you._

We were both on the ReactXP react-navigation thread where you mentioned:

> I've dropped all react native dev for now as the platform is still too unstable for me.

and

> My biggest gripe is all the random build errors like bundlejs and npm etc. Whether its CRA, RN or ReactXP. I'm going for PWA's for now.

I just wanted to hear how your experience has been with PWA compared to RN/ReactXP.

I am starting to get really annoyed by RN/ReactXP and the amount of work that needs to be done including: 

- Different routing (client/server side)
- No Server side rendering out of the box
- No SEO out of the box
- App stores processes
- Checking how things look on each platform device
- Dependency issues with node
- Breakages on upgrading package versions
- Client side token storage and security
- etc...

It's freaking crazy when all you want to do is build a single reusable codebase and all those things are obstacles.

PWA's that google is pushing on the other hand seem that you TRULY write it once (a responsive website) and it works everywhere a browser works (Apple is working on service workers it seems atm).

I'd like to hear your experience so far if you don't mind sharing from the transition you went through from RN/ReactXP to PWA's. The good, the bad, and the ugly?

**Comment by QuantumInformation** [comment]
Thanks for reaching out, you have exposed some of my pain points about RN.

My home site is a simple app built from my own open source libs. I'm deliberately keeping it ultra simple which has the side effect of it being ultra maintainable. It is built from [several](https://github.com/quantumjs/quantumjs.github.io/blob/master/package.json#L49) vanilla typescript components including an ultra-simple "[warp-router](https://github.com/quantumjs/warp-router)". I'll add PWD functionality to my own apps just-in-time when I need it.

Angular 5 looks interesting as it has "support" for PWA out the box, but I haven't used it.

**Comment by MovingGifts** [comment]
I see, so not too much in the PWA world yet. My backend uses php (Laravel), so I might look into getting it to run as a PWA, but I might have to scrap all my hard work so far in RN/ReactXP, and see what I can reuse and what to refactor. The pain points above are pretty serious ones (and there is more to add), so I might give PWA a shot for a month or so, see how speedy that development is without the above pain points.

Thanks for the speedy response btw! :)

**Comment by QuantumInformation** [comment]
The bigger pain point for me is the proliferation of js frameworks. I actually enjoying writing vanilla the most.

**Comment by MovingGifts** [comment]
I totally agree, just got to find something stable to stick to (which is why I use Laravel) which is php but is extremely stable. I think PWA will bridge the gap and I hope it grows from there since google is pushing it, but I'll need to look into screen transitions on web so they feel like native for the time being. I hope it works out well for us all :)

**Comment by QuantumInformation** [comment]
I'm rebuilding my site with https://github.com/vuetifyjs/pwa
